Control/Monad/State/Class.hs

```diff
@@ -192,3 +196,37 @@ instance MonadState s m => MonadState s (SelectT r m) where
     get = lift get
     put = lift . put
     state = lift . state
+
+-- | A helper type to decrease boilerplate when defining new transformer
+-- instances of 'MonadState'.
+--
+-- @since ????
+type LiftingState :: ((Type -> Type) -> Type -> Type) -> (Type -> Type) -> Type -> Type
+newtype LiftingState t m a = LiftingState (t m a)
+  deriving (Functor, Applicative, Monad, MonadTrans)
+
+instance (MonadState s m, Monoid w) => MonadState s (LiftingState (LazyRWS.RWST r w s') m) where
+  get = lift get
+  put = lift . put
+  state = lift . state
+
+instance (MonadState s m, Monoid w) => MonadState s (LiftingState (StrictRWS.RWST r w s') m) where
+  get = lift get
+  put = lift . put
+  state = lift . state
+
+instance (MonadState s m, Monoid w) => MonadState s (LiftingState (CPSRWS.RWST r w s') m) where
+  get = lift get
+  put = lift . put
+  state = lift . state
+
+instance MonadState s m => MonadState s (LiftingState (Lazy.StateT s') m) where
+  get = lift get
+  put = lift . put
+  state = lift . state
+
+instance MonadState s m => MonadState s (LiftingState (Strict.StateT s') m) where
+  get = lift get
+  put = lift . put
+  state = lift . state
```

Comment:
These instances could be unified via `MonadTrans`, like the instances for `LiftingAccum` and `LiftingSelect`. (The others unfortunately require more power; I believe `MonadTransControl` from `monad-control` would be sufficient but obviously that's not available here.)